r/R/compute.R

```diff
@@ -80,6 +80,36 @@ collect_arrays_from_dots <- function(dots) {
   ChunkedArray$create(!!!arrays)
 }
 
+#' @export
+quantile.ArrowDatum <- function(x,
+                                probs = seq(0, 1, 0.25),
+                                na.rm = FALSE,
+                                interpolation = c("linear", "lower", "higher", "nearest", "midpoint"),
+                                ...) {
+  if (inherits(x, "Scalar")) x <- Array$create(x)
+  assert_is(probs, c("numeric", "integer"))
+  assert_that(length(probs) > 0)
+  assert_that(all(probs >= 0 & probs <= 1))
+  if (!na.rm && TRUE %in% as.vector(unique(is.na(x)))) {
+    stop("Missing values not allowed if 'na.rm' is FALSE", call. = FALSE)
```

Comment:
Why doesn't this just return an Array of NA, like `median` does?

Is this a missing feature on the C++ side (needing a JIRA)?

Reply:
That's what `median` does. This is what `quantile` does when `na.rm = FALSE`.

`median` wraps this but handles the `na.rm = FALSE` case separately so this `stop()` will never run when `median` is called.

Reply:
Ah, gotcha, this is what R does. Gotta love the inconsistency.